### PR TITLE
PROD-1247 react timeSelect upgrade to react 14

### PR DESCRIPTION
**Pull request by reecemillsom**
#### What does this PR do? (please provide any background)

This PR upgrades react to react 14 and fixes the component to work with 14.
#### What tests does this PR have?

Unit tests run 'npm test' and make sure they all pass.
#### How can this be tested?

Run the example do npm start and make sure the page has no errors and times can be selected from the list.
#### Screenshots / Screencast

http://grab.by/NCEE
#### What gif best describes how you feel about this work?

---
- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.
#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** *
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

**Comment by djbeaumont** [comment]
Taking :two:

**Comment by Joezo** [comment]
Taking 1

File: test/testTimeSelect.jsx [hunk 1]
```diff
@@ -48,66 +44,65 @@ describe('TimeSelect', function() {
...
+        label: 'Time'
+      };
+    });
+
+    it('should render the reactIntl', function() {
+      var timeSelect = shallow(<TimeSelect />);
+      expect(timeSelect.find(ReactIntl.IntlProvider)).to.have.length(1);
     });
 
-    it('can be provided a default value as a Date instance', function() {
```
Comment (Joezo) [inline test/testTimeSelect.jsx:57]: this test doesn't appear to exist anymore :(

File: test/testTimeSelect.jsx [hunk 2]
```diff
@@ -48,66 +44,71 @@ describe('TimeSelect', function() {
...
+      expect(timeSelect.find(Input)).to.have.length(1);
+    });
 
-      assert.equal(renderOutput.props.name, 'foo');
-      assert.equal(renderOutput.props.className, 'baz');
-      assert.equal(renderOutput.props.label, 'bar');
+    it('should render the Input field with a type prop', function() {
+      var timeSelect = shallow(<TimeSelect {...props}/>);
+      expect(timeSelect.find(Input).props().type).to.equal(props.type);
+    });
```
Comment (djbeaumont) [inline test/testTimeSelect.jsx:75]: `type` is statically defined on the input so the way the test is defined is slightly confusing. Could you take `type` out of the props object here and instead do:

```
expect(timeSelect.find(Input).props().type).to.equal('select');
```

File: test/testTimeSelect.jsx [hunk 3]
```diff
@@ -48,66 +44,71 @@ describe('TimeSelect', function() {
...
-        assert.equal(option.props.children.type, ReactIntl.FormattedTime);
-      });
-
-      // FormattedTimes expect date instances as a starting point to be formatted
-      assert.deepEqual(renderOutput.props.children.map(function(option) {
-        var date = option.props.children.props.value;
-        return [ date.getHours(), date.getMinutes() ];
-      }), [ [10, 0], [10, 15], [10, 30], [10, 45], [11, 0], [11, 15] ]);
+      var timeSelect = shallow(<TimeSelect start={1000} end={1130} step={15} locale="en-US"/>);
+      expect(timeSelect.find(ReactIntl.FormattedTime)).to.have.length(6);
```
Comment (djbeaumont) [inline test/testTimeSelect.jsx:110]: This doesn't really seem to test the localisation. Is it possible to choose a locale that uses a different time format and test that the values are as expected?

**Comment by djbeaumont** [comment]
There's a warning of `Warning: require('react/addons') is deprecated. Access using require('react-addons-{addon}') instead` that would be nice to sort out as part of the move to React 0.14

**Comment by djbeaumont** [comment]
Will you update the version (v1.1.0?) and make a new tag/release on github?

**Comment by reecemillsom** [comment]
@djbeaumont i shall be eventually

**Comment by reecemillsom** [comment]
@djbeaumont removed the warning you received above

**Comment by Joezo** [comment]
Could you update the README to reflect the breaking change please? think it's the locales not being an array any more

**Comment by djbeaumont** [comment]
:+1:

File: README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # react-time-select
 
-[React](https://facebook.github.io/react/) Component to render a [React Bootstrap](https://react-bootstrap.github.io/) dropdown list pre-filled with localizable times formatted by [ReactIntl](http://formatjs.io/react/).
+[React](https://facebook.github.io/react/) Component to render a [React Bootstrap](https://react-bootstrap.github.io/) dropdown list pre-filled with localizable times formatted by [ReactIntl](http://formatjs.io/react/). Requires react v0.14.2.
```
Comment (Joezo) [inline README.md]: Could you make this v0.14+

**Comment by Joezo** [comment]
:+1:

**Pull request merged**